### DISCUSSION TITLE
Format directories to a single naming scheme

### DISCUSSION THREAD

**Discussion by apraga** [papis/papis#904] [Q&A] [answered]
Hi,

After merging 2 librairies and adding papers from zotero, there a 3 filenaming schemes in my library:
- author-title (ex: zook-a-robust-benchmark-for-detection-of-germline-large-deletions-and-insertions    )
- hash (?) (ex: ZZNUCRTP  )
- hash-author (?) (ex: a6677d773e52d4fc1773395273282bdd-bettio-cinzia-and-s  )
Is there a way for papis to rename folder and enforce a single format ?

Thanks,

**Comment by alexfikl** [comment]
There's no nice automatic way to do this at the moment. The command that should be able to do this is `papis mv`, but it's pretty bare (never actually used it :-s) with some improvements in #826.

I had the same problem at some point and wrote a little script to just go through all the documents and do a `shutil.move(new_dir, old_dir)` using `add-folder-name`. Ideally, we could teach `papis doctor` to do that :grin:

**Comment by apraga** [comment]
Well, `papis rename --folder-name "{doc[title]}" --all --batch` did the trick !
One last question : I have multiple authors in the author field. Is there a way to extract only the first author from thar in `--folder-name` ?

**Reply by alexfikl** [reply, marked as answer]
You can't extract much from the author field in a robust way, I'm afraid. If it's just a single document, you could do something like
```
papis rename --folder-name "{doc[author]:5.10S}"
```
where "5.10S" just slices into the author string, i.e. `doc[author][5:10]`. You could use the `author_list` entry instead if you know that all documents have a first author
```
papis rename --folder-name "{doc[author_list][0][family]}"
```

Besides these, you're only hope is probably switching to the `jinja2` formatter. That has a lot of little helper functions: https://jinja.palletsprojects.com/en/3.1.x/templates/#builtin-filters. We can also always add more custom ones if they're generally useful.
